src/cc_catalog_airflow/dags/provider_api_scripts/test_metropolitan_museum_of_art.py

```diff
@@ -0,0 +1,265 @@
+            ),
+        'culture': 'Japan',
+        'objectDate': 'late 17th century',
+        'medium': 'Hanging scroll; ink and color on silk',
+    }
+    exact_meta_data = {
+        'accession_number': '36.100.45',
+        'classification': 'Paintings',
+        'credit_line': (
+            'The Howard Mansfield Collection, Purchase, Rogers Fund, 1936'
+            ),
+        'culture': 'Japan',
+        'date': 'late 17th century',
+        'medium': 'Hanging scroll; ink and color on silk',
+    }
+    r = requests.Response()
+    r.status_code = 200
+    r.json = MagicMock(return_value=exact_response)
+    with patch.object(
+        mma.delayed_requester,
+        'get',
+        return_value=r
+    ) as mock_get:
+        response = mma._get_response_json(None, '', retries=2)
+        meta_data = mma._create_meta_data(response)
+
+    assert exact_meta_data == meta_data
+
+
+def test_process_image_data_handles_example_dict():
+    with open(os.path.join(RESOURCES, 'sample_image_data.json')) as f:
+        image_data = json.load(f)
+
+    with patch.object(
+        mma.image_store,
+        'add_item',
+        return_value=1
+    ) as mock_add:
+        mma._process_image_data(image_data)
+
+    mock_add.assert_called_once_with(
+        foreign_landing_url=(
+            "https://wwwstg.metmuseum.org/art/collection/search/45734"
+            ),
+        image_url=(
+            "https://images.metmuseum.org/CRDImages/as/original/DP251139.jpg"
+            ),
+        thumbnail_url=(
+            "https://images.metmuseum.org/CRDImages/as/web-large/DP251139.jpg"
+            ),
+        license_="cc0",
+        license_version="1.0",
+        foreign_identifier=45734,
+        creator="Kiyohara Yukinobu",
+        title="Quail and Millet",
+        meta_data={
+            "accession_number": "36.100.45",
+            "classification": "Paintings",
+            "credit_line": (
+                "The Howard Mansfield Collection, Purchase, Rogers Fund, 1936"
+                ),
+            "culture": "Japan"
+        }
+    )
+
+def test_get_data_for_image_with_none_response():
+    with patch.object(
+            mma.delayed_requester,
+            'get',
+            return_value=None
+    ) as mock_get:
+        with pytest.raises(Exception):
+            assert mma._get_data_for_image(10)
+
+    assert mock_get.call_count == 6
+
+def test_get_data_for_image_with_non_ok():
+    r = requests.Response()
+    r.status_code = 504
+    r.json = MagicMock(return_value={})
+    with patch.object(
+            mma.delayed_requester,
+            'get',
+            return_value=r
+    ) as mock_get:
+        with pytest.raises(Exception):
+            assert mma._get_data_for_image(10)
+
+    assert mock_get.call_count == 6
+
+def test_get_data_for_image_returns_response_json_when_all_ok(monkeypatch):
+    with open(os.path.join(RESOURCES, 'sample_response_without_additional.json')) as f:
```

Comment:
While long lines are allowed in the test file in the case of long strings, please avoid them in situations like this where there's no chance of making a mistake on string formatting.
```suggestion
    with open(
            os.path.join(RESOURCES, 'sample_response_without_additional.json')
    ) as f:
```